runtime/lua/vim/treesitter.lua

```diff
@@ -466,4 +466,27 @@ function M.foldexpr(lnum)
   return M._fold.foldexpr(lnum)
 end
 
+--- Get highlight information from parser.
+---
+---@param parser vim.treesitter.LanguageTree
+---@return vim.treesitter.highlighter.Iter.ret[]
+function M.get_highlight(parser)
+  local highlights = {}
+  parser:for_each_tree(function(tree, ltree)
+    local source = ltree:source()
+    local hl_query = M.query.get(ltree:lang(), 'highlights')
+
+    if not hl_query then
+      return
+    end
+
+    local highlighter = require('vim.treesitter.highlighter')
+    for info in highlighter._iter_highlight_info(hl_query, tree:root(), source, 0, -1) do
+      info.range = { M._range.unpack4(info.range) }
```

Comment:
`range` declared as Range6 but the content is Range4. Also it would be nice to add some tests for the fields in `vim.treesitter.highlighter.Iter.ret`.